### PR TITLE
testutil/compose: add lodestar

### PR DESCRIPTION
**Pull request by dB2510**
Adds support for lodestar VC to compose tool. Also updates beaconmock spec to be interoperable with all the validator clients. Minimal preset enables interoperable devnet testing between validator clients ([refer](https://github.com/ethereum/consensus-specs/blob/dev/presets/README.md)) which is supported by all the clients.

category: feature
ticket: #1897

### CODE REVIEW

File: testutil/compose/static/lodestar/run.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+mkdir /tmp/testnet/
+curl "http://${NODE}:3600/eth/v1/config/spec" | jq -r .data | yq -P > /tmp/testnet/config.yaml
+
+for f in /compose/"${NODE}"/validator_keys/keystore-*.json; do
+    echo "Importing key ${f}"
+
+    # Import keystore with password.
+    node /usr/app/packages/cli/bin/lodestar validator import \
+        --network="dev" \
+        --presetFile="/tmp/testnet/config.yaml" \
```
Comment (nflaig) [inline testutil/compose/static/lodestar/run.sh:19]: Only the network is relevant here as it is used to determine to which folder to store the keystores, `--presetFile` and `--paramsFile` can be removed.

File: testutil/compose/static/lodestar/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# TODO(dhruv): use official release version tag.
+FROM chainsafe/lodestar:next
```
Comment (nflaig) [inline testutil/compose/static/lodestar/Dockerfile:2]: [v1.8.0-rc.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.8.0-rc.0) is already released, we should have a stable release in ~2 days if there are no issues during rc testing phase

File: testutil/random.go [hunk 3]
```diff
@@ -656,27 +657,27 @@ func RandomCheckpoint() *eth2p0.Checkpoint {
 }
 
 func RandomEpoch() eth2p0.Epoch {
-	return eth2p0.Epoch(rand.Uint64())
+	return eth2p0.Epoch(rand.Int63n(int64(math.Pow(2, 53))))
 }
 
 func RandomSlot() eth2p0.Slot {
-	return eth2p0.Slot(rand.Uint64())
+	return eth2p0.Slot(rand.Int63n(int64(math.Pow(2, 53))))
```
Comment (dB2510) [inline testutil/random.go:664]: Note: Had to use `2^53` as boundary for these random values to resolve errors from lodestar.
image

File: testutil/compose/static/lodestar/run.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+node /usr/app/packages/cli/bin/lodestar validator \
+    --network="dev" \
+    --presetFile="/tmp/testnet/config.yaml" \
+    --paramsFile="/tmp/testnet/config.yaml" \
+    --metrics=true \
+    --metrics.address="0.0.0.0" \
+    --metrics.port=5064 \
+    --beaconNodes="http://$NODE:3600" \
+    --distributed
```
[the remark this answers is not in the document]
Comment (dB2510) [reply testutil/compose/static/lodestar/run.sh:33]: will avoid debug for now as it will be too verbose, will add if some issue pops up

**Review by corverroos — APPROVED** [review]

**Pull request merged**